src/elisa/models/bslog.py

```diff
@@ -0,0 +1,107 @@
+import jax.numpy as jnp
+import jax.random as random
+import numpy as np
+from jax import lax
+from numpyro.distributions import (
+    Distribution,
+    constraints,
+)
+from numpyro.distributions.distribution import (
+    Distribution,
+)
+from numpyro.distributions.util import (
+    promote_shapes,
+    validate_sample,
+)
+
+
+# # Bi-Symmetric log transformation
+# # https://iopscience.iop.org/article/10.1088/0957-0233/24/2/027001
+def BiSymmetricLog(
+    x: np.float64, C: np.float64 = 1 / np.log(10)
+) -> np.float64:
+    return np.sign(x) * np.log10(1 + np.abs(x / C))
+
+
+def InverseBiSymmetricLog(
+    y: np.float64, C: np.float64 = 1 / np.log(10)
+) -> np.float64:
+    return np.sign(y) * C * (-1 + np.power(10, np.abs(y)))
+
+
+class BiSymmetricLogUniform(Distribution):
+    arg_constraints = {
+        'low': constraints.dependent,
+        'high': constraints.dependent,
+    }
+    reparametrized_params = ['low', 'high']
+    pytree_data_fields = ('low', 'high', '_support')
+
+    def __init__(self, low=-1.0, high=1.0, *, validate_args=None):
+        self._c = 1 / jnp.log(10)
+        self.low, self.high = promote_shapes(low, high)
+        batch_shape = lax.broadcast_shapes(jnp.shape(low), jnp.shape(high))
+        self._support = constraints.interval(low, high)
+        super().__init__(batch_shape, validate_args=validate_args)
+
+    @constraints.dependent_property(is_discrete=False, event_dim=0)
+    def support(self):
+        return self._support
+
+    def sample(self, key, sample_shape=()):
+        log_low = self.bsLog(self.low)
+        log_high = self.bsLog(self.high)
+        shape = sample_shape + self.batch_shape
+        return self.bsLog_inverse(
+            random.uniform(key, shape=shape, minval=log_low, maxval=log_high)
+        )
+
+    @validate_sample
+    def log_prob(self, value):
+        shape = lax.broadcast_shapes(jnp.shape(value), self.batch_shape)
+        return -jnp.broadcast_to(self.bsLog(self.high - self.low), shape)
+
+    def cdf(self, value):
+        log_value = self.bsLog(value)
+        log_low = self.bsLog(self.low)
+        log_high = self.bsLog(self.high)
+        cdf = (log_value - log_low) / (log_high - log_low)
+        return jnp.clip(cdf, a_min=0.0, a_max=1.0)
+
+    def icdf(self, value):
+        log_value = self.bsLog(value)
+        log_low = self.bsLog(self.low)
+        log_high = self.bsLog(self.high)
+        return log_low + log_value * (log_high - log_low)
+
+    @property
+    def mean(self):
+        log_low = self.bsLog(self.low)
+        log_high = self.bsLog(self.high)
+        return self.bsLog_inverse(log_low + (log_high - log_low) / 2.0)
+
+    @property
+    def variance(self):
```

Comment:
Make sure the mean and variance are correct. Note that this can be tested against samples' mean and variance.